src/utils/mergeConfigurations.js

```diff
@@ -6,16 +6,20 @@ export default ({ globalConfiguration = {}, localConfiguration = {} }) => {
     version: globalVersion,
     accessToken,
     getAccessToken,
+    customHeaders: globalHeaders,
   } = globalConfiguration; // only destruct what we might need on request level
 
-  const { version: localVersion, fetchAll } = localConfiguration;
+  const { version: localVersion, fetchAll, customHeaders: localHeaders } = localConfiguration;
 
   const plugins = mergePlugins(globalConfiguration.plugins, localConfiguration.plugins);
 
+  const customHeaders = { ...globalHeaders, ...localHeaders };
```

Comment:
Are we sure that these will always be objects?  Won't we get an error here because you're spreading a non-iterable object?

Reply:
When it undefined it will defaul to empty object in createRequestHandlers but perhaps some added validation needs to applied here. 👍